build-system/tasks/karma.conf.js

```diff
@@ -251,7 +251,7 @@ module.exports = {
   },
 
   singleRun: true,
-  browserDisconnectTimeout: 10000,
+  browserDisconnectTimeout: 4 * 60 * 1000,
```

Comment:
This is going from 10 seconds to 4 minutes. From the [documentation](http://karma-runner.github.io/3.0/config/configuration-file.html), it looks like this won't greatly increase out test running time in the case of failures, while potentially making the framework more resilient. Worth trying.